### PR TITLE
Add Recipe info logic

### PR DESCRIPTION
**Pull request by hoapq-1005**
## Related Tickets

- [#25996](https://edu-redmine.sun-asterisk.vn/issues/25996)

## What's this PR do ?

- [x] Add Recipe Info Logic
- [x] Fix rxswift disposable create

## Library
*(List gem, library third party add new)*

## Impacted Areas in Application
*(List features, api, models or services that this PR will affect)*

- Scene/RecipeInfo
- Extensions/UIViewController+

## Checklist

- [x] It was tested in local success ?
- [x] Fill link PR into ticket and the opposite
- [x] Note reason, scope of influence, solution into ticket
- [x] Validate UI/Model/API

## Screenshot
ezgif com-video-to-gif (5)

### CODE REVIEW

File: Delicious/Scenes/RecipeInfo/RecipeInfoViewController.swift [hunk 1]
```diff
@@ -115,10 +136,93 @@ final class RecipeInfoViewController: UIViewController, BindableType {
...
+            shoppingViewBottomConstraint.constant = 16
+            UIView.animate(withDuration: 0.25) {
+                self.view.layoutIfNeeded()
+            }
+        } else {
+            shoppingViewBottomConstraint.constant = -150
+            UIView.animate(withDuration: 0.25) {
+                self.view.layoutIfNeeded()
+            }
+        }
```
Comment (dongtg-1485) [inline Delicious/Scenes/RecipeInfo/RecipeInfoViewController.swift]:
```suggestion
        shoppingViewBottomConstraint.constant = status ? -150 : 16
        UIView.animate(withDuration: 0.25) {
                self.view.layoutIfNeeded()
        }
```

File: Delicious/Scenes/RecipeInfo/RecipeInfoViewController.swift [hunk 2]
```diff
@@ -115,10 +136,93 @@ final class RecipeInfoViewController: UIViewController, BindableType {
...
+        output.shoppingButtonHidden
+            .drive(isShoppingButtonHidden)
+            .disposed(by: rx.disposeBag)
+        output.isFavorited
+            .drive(isFavorited)
+            .disposed(by: rx.disposeBag)
+        output.tapShopingList
+            .do(onNext: { [weak self] _ in
+                self?.loadShopingListTrigger.onNext(())
+                self?.showAlertConfirm(title: "Success",
```
Comment (dongtg-1485) [inline Delicious/Scenes/RecipeInfo/RecipeInfoViewController.swift]: Show alert trong navigator

File: Delicious/Scenes/RecipeInfo/RecipeInfoViewController.swift [hunk 3]
```diff
@@ -115,10 +136,86 @@ final class RecipeInfoViewController: UIViewController, BindableType {
...
+                default:
+                    return ""
+                }
+            })
+
+        let input = RecipeInfoViewModel.Input(
+            loadTrigger: Driver.just(()),
+            reloadTrigger: refreshControl.refreshTrigger,
+            favoriteTrigger: favoriteButton.rx
+                                           .tap
```
Comment (sonnd-1148) [inline Delicious/Scenes/RecipeInfo/RecipeInfoViewController.swift]: thêm throtlet để tránh trường họp bấm quá nhanh nhé

**Review by sonnd-1148 — APPROVED** [review]

File: Delicious/Scenes/RecipeInfo/RecipeInfoViewController.swift (hunk 3 above)
Comment (hoapq-1005) [reply Delicious/Scenes/RecipeInfo/RecipeInfoViewController.swift]: Em thêm rồi ạ!

**Review by datdt-1935 — APPROVED** [review]

**Pull request merged**